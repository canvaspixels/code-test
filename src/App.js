import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import getChatLog from './service';
import MessageList, { messageListProps } from './MessageList';

import './App.css';

class App extends Component {
  componentDidMount() {
    this.props.getChatLog();
  }
  render() {
    const { messages } = this.props;
    return (
      <MessageList messages={messages} />
    );
  }
}

App.propTypes = messageListProps;
App.defaultProps = {
  messages: []
};

const mapStateToProps = state => {
  return {
    messages: state.messages
  };
};

const mapDispatchToProps = dispatch => bindActionCreators({ getChatLog }, dispatch);

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App);
