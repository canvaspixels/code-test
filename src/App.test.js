import React from 'react';
import { shallow } from 'enzyme';
import ReactDOM from 'react-dom';
import App from './App';
const getStubMessages = () => ([]);

describe('App', () => {
  it('matches snapshot', () => {
    const component = shallow(<App.WrappedComponent messages={getStubMessages()} />);

    expect(component).toMatchSnapshot();
  });

  it('should render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App.WrappedComponent getChatLog={() => {}} />, div);
  });

  it('should call getChatLog on mount', () => {
    const div = document.createElement('div');
    const getChatLogStub = jest.fn();
    ReactDOM.render(<App.WrappedComponent getChatLog={getChatLogStub} />, div);
    expect(getChatLogStub).toHaveBeenCalled();
  });
});
