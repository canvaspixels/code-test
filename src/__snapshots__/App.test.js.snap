exports[`App matches snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <MessageList
    messages={Array []} />,
  "nodes": Array [
    <MessageList
      messages={Array []} />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <App
        messages={Array []} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": App {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "messages": Array [],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <MessageList
          messages={Array []} />,
        "_debugID": 2,
        "_renderedOutput": <MessageList
          messages={Array []} />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <App
    messages={Array []} />,
}
`;
